### ISSUE TITLE
Accessibility: governance intro

### ISSUE THREAD

**Issue by Abolade017** [jupyterhub/outreachy#102]
### What page is this for?

https://github.com/jupyterhub/outreachy/issues/38#issuecomment-1276035815

### WAVE accessibility report

I evaluated the web accessibility of [https://jupyter.org/governance/intro.html](url) page. The evaluation is the following:
![jupyter governance](https://user-images.githubusercontent.com/70269933/195323133-0a17c184-6a64-432c-a03f-026fde110f38.JPG)

### **Evaluation**

- **Empty form label (5) :** A form label is present, but does not contain any content. A `<label>` element that is associated to a form control but does not contain text will not present any information about the form control to the user. It can be fixed by ensuring that the form label contains text that describes the function of the associated form control. Labels are not required for image, submit, reset, button, or hidden form controls. If a `label` is not necessary visually, a descriptive `title attribute` may be added to the form control.
- **Multiple form labels(2) :** Two or more `<label>`s are associated to a single `<input>` (except types of `image`, `submit`, `reset`, `button`, or `hidden`), `<select>`, or `<textarea>`. A form control should have at most one associated label element. If more than one label element is associated to the control, assistive technology may not read the appropriate label. It can be fixed by ensuring that at most one label element is associated to the form control. If multiple form labels are necessary, use `aria-labelledby`.
- **language (1) :** The `<html lang>` attribute is missing or is empty, or a` lang` attribute value is not a valid language identifier. Identifying the language of the page or page elements allows screen readers to read the content in the appropriate language. It also facilitates automatic translation of content. It can be identifying the document language using the `<html lang>` attribute with a valid value (e.g., `<html lang="en">`). Ensure that all `lang` attribute values are valid.
- **Empty button (1) :** A `<button>` element is present that contains no text content (or alternative text), or an `<input type="submit">`, `<input type="button">`, or `<input type="reset">` has an empty or missing value attribute. When navigating to a button, descriptive text must be presented to screen reader users to indicate the function of the button. It can be fixed by Placing text content within the `<button>` element or give the `<input>` element a value attribute.
- **Redundant link (1) :** Two adjacent links go to the same `URL`. When adjacent links go to the same location (such as a linked product image and an adjacent linked product name that go to the same product page) this results in additional navigation and repetition for keyboard and screen reader users. If possible,it can be fixed by combining the redundant links into one link and remove any redundant text or alternative text (for example, if a product image and product name are in the same link, the image can usually be given `alt=""`).
- **Linked imaged with alternative text (1) :** An image element has non-empty alternative text, is within a link, and no other text (or images with alternative text) is present within the link. Including appropriate alternative text on an image within a link ensures that the function and purpose of the link and the content of the image is available to screen reader users or when images are unavailable. It can be fixed by ensuring that the alternative text presents the content of the image and/or the function of the link. If the full content and function of the link is presented in text within the link (an image and a text caption both within the same link, for example), then the image should generally be given empty/null alternative text (`alt=""`) to avoid redundancy.
- **Form label (2) :** A `<label>` element is present and properly associated to `<input>` (except types of `image`, `submit`, `reset`, `button`, or `hidden`), `<textarea>`, or `<select>` element. A properly associated form label is presented to a screen reader user when the form control is accessed. Additionally, a label can be clicked with the mouse to set focus to the form control. It can be fixed by ensuring that the label is accurate, descriptive, succinct, and that it is associated with the correct form control element.
- **Heading level element (6) :** There are 3 first level heading elements (`<h1>` element) and 3second level heading elements(`<h2>`) are present. Headings facilitate page navigation for users of assistive technologies. They also provide semantic and visual meaning and structure to the document. First level headings should contain the most important heading(s) on the page (generally the document title). Heading level element issue can be fixed by ensuring that the text in question is truly a heading and that it is structured correctly in the page outline.
- **Unordered list (13) :** A `<ul>` element is present. It can be fixed by ensuring that an unordered (bulleted) list is appropriate for the context. If list items are sequential or numbered, an ordered list (`<ol>`) is likely more appropriate.
- **Navigation (1) :** A `<nav>` element or role="navigation" is present. The navigation identifies a section of navigation links and can facilitate page semantics and navigation. It can be fixed by ensuring the element defines page navigation. Multiple navigation elements on one page can be differentiated with `ARIA` labels.
- **Main content (1) :** A `<main>` element or `role="main"` is present. The `<main>` element or `role="main"` attribute identifies the main content for the page. This facilitate page semantics and navigation. It can be fixed by ensuring the element surrounds and defines page main content.
- **Footer (1) :** A `<footer>` element or `role="contentinfo"` is present. Footers identify a footer for the page or a page section. It typically identifies authorship, related links, copyright date, or other footer content. Footers facilitate page semantics and navigation. It can be fixed by ensuring that the element surrounds and defines page or page section footer content.
- **ARIA (5) :** An `ARIA` role, state, or property is present, excluding landmark roles, `aria-labelledby`, or` aria-describedby` which are distinct WAVE items. `ARIA` provides enhanced semantics and accessibility for web content. It can be fixed by ensuring that the `ARIA` role, state, or property is used correctly. Use standard `HTML` accessibility features when possible. Be aware that support for `ARIA` is limited in older browsers and assistive technologies.
- **ARIA labels (7) :** An `aria-label` or `aria-labelledby` attribute is present. ARIA labels define accessible names to be read by screen readers for interface elements. They may be used when `HTML` associations (label, alternative text, etc.) are not sufficient. It can be fixed by ensuring that the `aria-label` value or element referenced by `aria-labelledby` provides a descriptive, accurate label. When possible, use standard HTML `<label>` or other markup to make the association.

### The HTML element to be changed

```html
1. Button is not having an inner text:
<button onclick="toggleFullScreen()" class="headerbtn" data-toggle="tooltip" data-placement="bottom" title="" data-original-title="Fullscreen mode">
<span class="headerbtn__icon-container">
  <i class="fas fa-expand"></i>
  </span>
</button>
2. Missing lang attribute in <html style="--pulse-color: #fffc0230; margin-left: 380px;">.
3. I am changing <h1 class="site-logo" id="site-title">
Project Jupyter Governance
</h1> because a page should contain only one heading level 1 element and it should be in the <header> element.
4. I will change the second <footer> from this:
<footer class="footer-article noprint">
<!-- Previous / next buttons-->
<div class="prev-next-area"><a class="right-next" id="next-link" href="conduct/code_of_conduct.html" title="next page"><div class="prev-next-info"><p class="prev-next-subtitle">
next
</p><p class="prev-next-title">
Project Jupyter Code of Conduct
</p></div><i class="fas fa-angle-right"></i></a></div></footer>
</div></div><div class="footer-content row">
Footer<footer class="col footer"><p>
By The Jupyter Book community
<br></p><div class="extra_footer"><p xmlns:dct="http://purl.org/dc/terms/" xmlns:vcard="http://www.w3.org/2001/vcard-rdf/3.0#"><a rel="license" href="http://creativecommons.org/publicdomain/zero/1.0/">
Linked image with alternative text<img src="https://licensebuttons.net/p/zero/1.0/88x31.png" style="border-style: none;" alt="CC0">
</a><br>
To the extent possible under law,
<a rel="dct:publisher" href="https://jupyter.org"><span property="dct:title">
Project Jupyter
</span></a>
has waived all copyright and related or neighboring rights to
<a href="https://github.com/jupyter/governance"><span property="dct:title">
Project Jupyter Governance documents
</span></a>
. This work is published from:
<span property="vcard:Country" datatype="dct:ISO3166" content="US" about="https://jupyter.org">
United States
</span>
.
</p></div><p></p></footer> because a page should have a single footer.

5. i will change the <label> element in these:
    <input class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" type="checkbox">
<label for="toctree-checkbox-1"><i class="fas fa-chevron-down"></i></label>
<input class="toctree-checkbox" id="toctree-checkbox-2" name="toctree-checkbox-2" type="checkbox">
<label for="toctree-checkbox-2"><i class="fas fa-chevron-down"></i></label>
<input class="toctree-checkbox" id="toctree-checkbox-3" name="toctree-checkbox-3" type="checkbox">
<label for="toctree-checkbox-3"><i class="fas fa-chevron-down"></i></label>
6. I will change <label> element from this:
   <label for="__page-toc" class="headerbtn headerbtn-page-toc"><span class="headerbtn__icon-container"><i class="fas fa-list"></i></span></label>
```

### Your proposed HTML (or CSS), after change

```html
1. I will add an inner text to <button onclick="toggleFullScreen()" class="headerbtn" data-toggle="tooltip" data-placement="bottom" title="" data-original-title="Fullscreen mode">
<span class="headerbtn__icon-container">
  <i class="fas fa-expand"></i>
  </span>
</button>
2. I will include lang="en" to  <html style="--pulse-color: #fffc0230; margin-left: 380px;" lang="en">
3. I will change it to <h2 class="site-logo" id="site-title">
Project Jupyter Governance
</h2>
4. i will change the second footer to section element like this:
<footer class="footer-article noprint">
<!-- Previous / next buttons-->
<div class="prev-next-area"><a class="right-next" id="next-link" href="conduct/code_of_conduct.html" title="next page"><div class="prev-next-info"><p class="prev-next-subtitle">
next
</p><p class="prev-next-title">
Project Jupyter Code of Conduct
</p></div><i class="fas fa-angle-right"></i></a></div></footer>
</div></div><div class="footer-content row">
<section class="col footer"><p>
By The Jupyter Book community
<br></p><div class="extra_footer"><p xmlns:dct="http://purl.org/dc/terms/" xmlns:vcard="http://www.w3.org/2001/vcard-rdf/3.0#"><a rel="license" href="http://creativecommons.org/publicdomain/zero/1.0/">
<img src="https://licensebuttons.net/p/zero/1.0/88x31.png" style="border-style: none;" alt="CC0">
</a><br>
To the extent possible under law,
<a rel="dct:publisher" href="https://jupyter.org"><span property="dct:title">
Project Jupyter
</span></a>
has waived all copyright and related or neighboring rights to
<a href="https://github.com/jupyter/governance"><span property="dct:title">
Project Jupyter Governance documents
</span></a>
. This work is published from:
<span property="vcard:Country" datatype="dct:ISO3166" content="US" about="https://jupyter.org">
United States
</span>
.
</p></div><p></p></section>
</footer>
5. i will change the code into this:
<div style="display:flex;justify-content:space-between;"><div><input class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" type="checkbox" aria-label="toctree-checkbox-1">
<label for="toctree-checkbox-1">toctree-checkbox</label></div>
<div><i class="fas fa-chevron-down"></i></div></div>
<div style="display:flex;justify-content:space-between;"><div><input class="toctree-checkbox" id="toctree-checkbox-2" name="toctree-checkbox-2" type="checkbox" aria-label="toctree-checkbox-2">
<label for="toctree-checkbox-2">toctree-checkbox</label></div>
<div><i class="fas fa-chevron-down"></i></div></div>
<div style="display:flex;justify-content:space-between;"><div><input class="toctree-checkbox" aria-label="toctree-checkbox-3" id="toctree-checkbox-3" name="toctree-checkbox-3" type="checkbox">
<label for="toctree-checkbox-1">toctree-checkbox</label></div>
<div><i class="fas fa-chevron-down"></i></div></div>
6. I will change <label> to <div> like this:
  <div  class="headerbtn headerbtn-page-toc"><span class="headerbtn__icon-container"><i class="fas fa-list"></i></span></div>
```

**Comment by minrk** [comment]
Thanks for hte suggestions! Can you explain the use of `style="display:flex;justify-content:space-between;"` in your divs replacing labels?

**Comment by Abolade017** [comment]
There is an error saying there is an empty label form, so when I checked the <label> element and I realized that there is an `<i>` element inside the label but there is not text between the `<label><label>`tag,  so I think of wrapping the `<label>` and `<i class="fas fa-chevron-down"></i>` element in the `<div>` and give it a display: flex to put them on the same line and the justify-content: space-between to give space between the label and the icon. So therefore, I just change the position of the `<label>` so that the label name will remove the empty label error.

**Comment by Abolade017** [comment]
There is an error saying there is an empty label form, so when I checked the <label> element and I realized that there is an `<i>` element inside the label but there is not text between the `<label><label>`tag,  so I think of wrapping the `<label>` and `<i class="fas fa-chevron-down"></i>` element in the `<div>` and give it a `display: flex;` to put them on the same line and the `justify-content: space-between;` to give space between the label and the icon. So therefore, I just change the position of the `<label>` so that the label name will remove the empty label error.

**Comment by sgibson91** [comment]
Thank you for your contribution! Since the contribution period is now over, we will close this issue.
